src/quartz/parser/qasm_parser.h

```diff
@@ -24,9 +24,26 @@ bool is_gate_string(const std::string &token, GateType &type);
 
 std::string strip(const std::string &input);
 
+class ParamParser {
+ public:
+  ParamParser(Context *ctx, bool symbolic_pi)
+      : ctx_(ctx), symbolic_pi_(symbolic_pi) {}
+
+  int parse(std::string &token);
+
+ private:
+  int parse_deg(bool negative, ParamType p);
+  int parse_rad(bool negative, ParamType num, ParamType denom);
```

Comment:
I didn't understand the difference between `parse_deg` and `parse_rad`. If I understand correctly, this is not degree/radian, but `parse_number` and `parse_pi_expression`? It's probably also better to add a comment to show the results are `p` and `num * PI / denom` correspondingly.

Reply:
I've tried clarifying this in one of the new commits.